### PR TITLE
Add new choose_response policy

### PR DESCRIPTION
**Pull request by Neverlord**
The choose policy simply picks the first valid result for a fan-out request and ignores all other responses.

### CODE REVIEW

File: libcaf_core/caf/policy/choose_response.hpp [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    CAF_LOG_TRACE(CAF_ARG(ids_));
+    using factory = detail::choose_response_factory<std::decay_t<F>>;
+    auto pending = std::make_shared<size_t>(ids_.size());
+    auto fw = factory::make(pending, std::forward<F>(f));
+    auto gw = make_error_handler(std::move(pending), std::forward<G>(g));
+    for (auto id : ids_) {
+      typename Self::accept_one_cond rc;
+      auto fcopy = fw;
+      auto gcopy = gw;
+      self->varargs_receive(rc, id, fcopy, gcopy);
```
[the remark this answers is not in the document]
Comment (Neverlord) [reply libcaf_core/caf/policy/choose_response.hpp]: `varargs_receive` takes the function objects by lvalue reference *and then moves them*. Terrible API, but it was never meant for use outside of `blocking_actor`. Fixing that API is a story for another PR, though. 🙂

[the remark this answers is not in the document]
Comment (josephnoir) [reply libcaf_core/caf/policy/choose_response.hpp]: Ah, thank you for the explanation!

**Comment by Neverlord** [comment]
> I am a bit confused about the name

I agree, the names are a bit odd at the moment. So what about renaming the two policies like this:

`fan_in_responses => select_all`
`choose_response => select_any`

?

**Comment by josephnoir** [comment]
Sounds better to me. I guess there is no use case for something like select the last result? Meaning `_any` as a suffix is fine because we are not going to have `_last` as well.

**Comment by jakobod** [comment]
I think the _any suffix fits, but the _all suffix doesn't. I just read the comment in `fan_in_responses.hpp` and that says: 
```cpp
/// Enables a `response_handle` to fan-in multiple responses into a single
/// result
```
Coming from there: I think something along the lines of `select_multiple` is clearer since it doesn't select *all*, but multiple responses.

**Comment by Neverlord** [comment]
Seems like I didn't just mess up the name but also the documentation. 😉

The `fan_in_responses` policy has all-or-nothing semantics. If you do a fan-out request and then use this policy, you'll either get *all* results or a *none* (i.e. an error).

**Comment by jakobod** [comment]
Then`select_all` and `select_any` fit in my opinion. :+1:

**Pull request merged**